intezer_sdk/endpoint_analysis.py

```diff
@@ -62,3 +90,144 @@ def _init_sub_analyses(self):
                                               api=self._api,
                                               verdict=sub_analysis['verdict'])
             self._sub_analyses.append(sub_analysis_object)
+
+    def _send_analyze_to_api(self, **additional_parameters) -> str:
+        try:
+            if not self._metadata_dir:
+                raise ValueError('Scan directory is not set')
+            if not os.path.isdir(self._metadata_dir):
+                raise ValueError('Scan directory does not exist')
+            if not os.path.isdir(self._files_dir):
+                raise ValueError('Files directory does not exist')
+            if not os.path.isdir(self._fileless_dir):
+                raise ValueError('Fileless directory does not exist')
+            if not os.path.isdir(self._memory_modules_dir):
+                raise ValueError('Memory modules directory does not exist')
+
+            self._scan_id, self.analysis_id = self._create_scan()
+
+            self.status = consts.AnalysisStatusCode.IN_PROGRESS
+            self._initialize_endpoint_api()
+
+            self._send_host_info()
```

Comment:
Add some debug logs between the steps

Reply:
there are info prints inside each function